### PR TITLE
Better sorting in admin

### PR DESCRIPTION
**Pull request by ar4s**
For better sorting in admin.

### CODE REVIEW

**Comment by xliiv** [comment]
:ok:

**Comment by mkurek** [comment]
models should be ordered first by manufacturer name, then by model name - is it working like that now?

**Comment by ar4s** [comment]
TODO: tests

File: src/ralph/admin/change_list.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                try:
+                    none, pfx, idx = p.rpartition('-')
+                    field_name = self.list_display[int(idx)]
+                    fields = self.get_ordering_from_related_model_admin(
+                        pfx, field_name
+                    )
+                    if fields:
+                        # replace one field by list of ordering fields
+                        ordering.remove('{}{}'.format(pfx, field_name))
+                        ordering = fields + ordering
```
Comment (mkurek) [inline src/ralph/admin/change_list.py]: what if before this loop ordering was `['a', 'b', 'c']` and here we want to replace `'b'` to `'b.1', 'b.2'`? As far as i understand this code you're changing sort order here. Shouldn't it be sth like 

```
index = ordering.index('{}{}'.format(pfx, field_name))
ordering[index:index+1] = fields
```

?

File: src/ralph/tests/models.py [hunk 2]
```diff
@@ -12,3 +12,14 @@
...
 class Foo(AdminAbsoluteUrlMixin, models.Model):
     bar = models.CharField('bar', max_length=50)
+
+
+class Manufacturer(models.Model):
+    name = models.CharField(max_length=50)
+    country = models.CharField(max_length=50)
+
+
+class Car(models.Model):
```
Comment (xliiv) [inline src/ralph/tests/models.py]: ?

File: src/ralph/admin/tests/tests_functional.py [hunk 3]
```diff
@@ -200,6 +203,100 @@ class FooAdmin(RalphAdmin):
...
+            request=request,
+            list_display=list_display
+        )
+        return cl.get_ordering(request, model.objects.all())
+
+    def test_sort_related(self):
+        ordering = self._get_ordering_list(
+            Car, CarAdmin, [1, 2], ['pk', 'name', 'manufacturer']
+        )
+        test_data = [
```
Comment (mkurek) [inline src/ralph/admin/tests/tests_functional.py]: consider using [ddt](http://ddt.readthedocs.org/en/latest/index.html) for such a test

File: src/ralph/admin/tests/tests_functional.py [hunk 4]
```diff
@@ -200,6 +203,100 @@ class FooAdmin(RalphAdmin):
...
+        cl = self._change_list_factory(
+            model=Car,
+            model_admin=CarAdmin,
+            request=request,
+            list_display=list_display
+        )
+        return cl.get_ordering(request, model.objects.all())
+
+    def test_sort_related(self):
+        ordering = self._get_ordering_list(
```
Comment (mkurek) [inline src/ralph/admin/tests/tests_functional.py:307]: this isn't probably needed anymore

File: src/ralph/admin/views/main.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                        pfx, field_name
+                    )
+                    # replace one field by list of ordering fields
+                    # or move field at the begining
+                    old_field = '{}{}'.format(pfx, field_name)
+                    ordering.remove(old_field)
+                    ordering = (fields or [old_field]) + ordering
+                except (IndexError, ValueError):
+                    raise
+                    continue
```
Comment (mkurek) [inline src/ralph/admin/views/main.py]: so raise or continue? :)

**Comment by ghost** [comment]
:ok:

**Comment by mkurek** [comment]
:ok:

**Pull request merged**